### DISCUSSION TITLE
Provide official guidance on how to handle gateway deletion when routes are still attached to it

### DISCUSSION THREAD

**Discussion by zhiying-lin** [kubernetes-sigs/gateway-api#3333] [General]
Hi,

GEP has not defined the gateway deletion behavior/UX when routes are still attached to it.

There are several approaches to handle this.
1. Blocking the gateway deletion via the validation webhook.
2. Blocking the gateway deletion by returning the reconcile error and deletion will get stuck.
3. Allow the gateway deletion, and update the route condition as "invalid gateway reference"

Should we provide some guidance/recommendations on the GEP? so that user can set the right expectation about the API.

Thanks,

**Comment by youngnick** [comment]
In general, we've tried to avoid recommending things that stop deliberate actions from being taken.

Deleting a Gateway is intended to be a destructive action, that should be controlled via outside controls like RBAC, Gitops or similar, rather than by the API itself.

Because of this, the only mandated behavior is what's in 3 currently.

We should probably have conformance tests that validate the expected behavior though, and clarify it since it's not clear.
